src/main/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMDirectives.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.vault.common.openam
+
+import akka.event.LoggingAdapter
+import spray.routing.Directives._
+
+object OpenAMDirectives {
+  val tokenFromCookie = cookie(OpenAMConfig.tokenCookie) map {
+    // return the content of the cookie
+    _.content
+  }
+
+  val tokenFromOptionalCookie = optionalCookie(OpenAMConfig.tokenCookie) map {
+    // when there is a cookie
+    _ map {
+      // return the content of the cookie
+      _.content
+    }
+  }
+
+  val commonNameFromCookie = tokenFromCookie map {
+    // get the common name from the token
+    case token => commonNameFromToken(token)
+  }
+
+  val commonNameFromOptionalCookie = tokenFromOptionalCookie map {
+    // when there is a cookie
+    _ map {
+      // get the common name from the token
+      case token => commonNameFromToken(token)
+    }
+  }
+
+  // Partially based off DebuggingDirectives.logRequest, sans magnets, with requestInstance instead of mapRequest
+  // TODO: Any way to make this single log parameter implicit?
+  // The magnet pattern seems to always have at least one other parameter
+  // http://spray.io/blog/2012-12-13-the-magnet-pattern/
+  def logOpenAMRequest(log: LoggingAdapter) = commonNameFromCookie flatMap { commonName =>
```

Comment:
Will make this part of DSDEV-1820, creating a "common" repo. For now they will stay forked.